spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jooq/JooqAutoConfiguration.java

```diff
@@ -105,15 +107,18 @@ public DefaultExecuteListenerProvider jooqExceptionTranslatorExecuteListenerProv
 
 		private final VisitListenerProvider[] visitListenerProviders;
 
+		private final TransactionListenerProvider[] transactionListenerProviders;
+
 		public DslContextConfiguration(JooqProperties properties,
 				ConnectionProvider connectionProvider, DataSource dataSource,
 				ObjectProvider<TransactionProvider> transactionProvider,
 				ObjectProvider<RecordMapperProvider> recordMapperProvider,
 				ObjectProvider<RecordUnmapperProvider> recordUnmapperProvider,
 				ObjectProvider<Settings> settings,
 				ObjectProvider<RecordListenerProvider[]> recordListenerProviders,
-				ExecuteListenerProvider[] executeListenerProviders,
-				ObjectProvider<VisitListenerProvider[]> visitListenerProviders) {
+				ObjectProvider<ExecuteListenerProvider[]> executeListenerProviders,
```

Comment:
That change in `ExecuteListenerProvider` is unrelated to the purpose of this PR. Please revert it so that this change is properly focused.